File: nodejs/database/models/doctor.js
'use strict';

module.exports = (sequelize, DataTypes) => {
    const doctor = sequelize.define('doctor', {
        id_doctor: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        name: {
            type: DataTypes.STRING(30),
        },
        surname: {
            type: DataTypes.STRING(30),
        },
        mail: {
            type: DataTypes.STRING(30),
        },
        password: {
            type: DataTypes.STRING(30),
        },
        is_active: {
            type: DataTypes.BOOLEAN,
            defaultValue: false
        },
        id_hospital: {
                type: DataTypes.INTEGER
        }
    }, {
        freezeTableName: true,
        underscored: true,
        name: {
            singular: 'doctor',
            plural: 'doctor'
        }
    });

    doctor.associate = models => {
        doctor.belongsTo(models.hospital, {
            foreignKey: 'id_hospital'
        });

        doctor.hasMany(models.request, {
            foreignKey: 'id_doctor',
            onDelete: 'cascade'
        });
    };

    return doctor;
};

File: python/src/models/doctor.py
from sqlalchemy.dialects.mysql import TINYINT
from sqlalchemy import Column, Integer, String, ForeignKey
from sqlalchemy.orm import relationship
from database import Base


class Doctor(Base):
    __tablename__ = "doctor"
    id_doctor = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
    name = Column(String(30), nullable=False)
    surname = Column(String(30), nullable=False)
    mail = Column(String(30), nullable=False)
    password = Column(String(30), nullable=False)
    is_active = Column(TINYINT(1), nullable=False, default=0)
    id_hospital = Column(Integer, ForeignKey("hospital.id_hospital"), nullable=False, ondelete="CASCADE")
    hospital = relationship("Hospital", back_populates="doctors")
    requests = relationship("Request", back_populates="doctor")
